### PR TITLE
perl.pod: use spaces instead of tabs for alignment

### PR DESCRIPTION
**Pull request by mauke**
On some platforms/formatters, the documentation overview in 'perldoc perl' renders badly (misaligned) because tabs are interpreted differently. This is presumably because some renderers output all of the leading spaces in a verbatim paragraph as-is, while others don't.

For example, https://perldoc.perl.org/perl currently shows for me:

                        |
                        v
    ...
    perlsyn         Perl syntax
    perldata                Perl data structures
    perlop          Perl operators and precedence
    perlsub         Perl subroutines
    perlfunc                Perl built-in functions
      perlopentut   Perl open() tutorial
      perlpacktut   Perl pack() and unpack() tutorial
    perlpod         Perl plain old documentation
    perlpodspec     Perl plain old documentation format specification
    perldocstyle    Perl style guide for core docs
    perlpodstyle    Perl POD style guide
    perldiag                Perl diagnostic messages
    perldeprecation     Perl deprecations
    perllexwarn     Perl warnings and their control
    perldebug               Perl debugging
    perlvar         Perl predefined variables
    ...
                        ^
                        |

All of the "Perl"s in the second column should be aligned like the "perldeprecation" line, which is the only one that uses spaces in the original POD document (the others use either one tab, two tabs, or a mix of spaces and tabs).

By using spaces everywhere we bypass the whole issue and get consistently aligned results, no matter how tabs are rendered.

### CODE REVIEW

**Comment by jkeenan** [comment]
+1

However, I notice that in both blead and your p.r., there are 3 entries under `=head2 Reference Manual`  in `pod/perl.pod` where the *leading* whitespace (as distinct from the *internal* whitespace which is the focus of your p.r.) is 6 characters rather than 4.
```
    perlfunc            Perl built-in functions$
      perlopentut       Perl open() tutorial$
      perlpacktut       Perl pack() and unpack() tutorial$
    perlpod             Perl plain old documentation$
...
    perlclass           Perl class syntax$
      perldbmfilter     Perl DBM filters$
$
    perlipc             Perl interprocess communication$
    perlfork            Perl fork() information$
    perlnumber          Perl number semantics$
$
    perlthrtut          Perl threads tutorial$
$
    perlport            Perl portability guide$
```
And, as you might notice above, these entries don't seem to be sorted alphabetically.  Could we fix these in this p.r., or would that be better done in separate patches?

**Comment by mauke** [comment]
`perlopentut` and `perlpacktut` are meant to be subsections of `perlfunc`. They provide a more in-depth explanation of a single function or group of functions.

Similarly, `perlfaq1` through `perlfaq9` are placed under `perlfaq`, and `perlothrtut` (when it still existed) was placed under `perlthrtut`.

`perldbmfilter` doesn't make sense as a subsection of `perlclass`, but before the latter was added to `perl.pod`, `perldbmfilter` was placed under `perltie`, which makes a lot more sense. Moving `perldbmfilter` to its previous location is the only change I'd make here.

**Comment by jkeenan** [comment]
> `perlopentut` and `perlpacktut` are meant to be subsections of `perlfunc`. They provide a more in-depth explanation of a single function or group of functions.
> 
> Similarly, `perlfaq1` through `perlfaq9` are placed under `perlfaq`, and `perlothrtut` (when it still existed) was placed under `perlthrtut`.
> 
> `perldbmfilter` doesn't make sense as a subsection of `perlclass`, but before the latter was added to `perl.pod`, `perldbmfilter` was placed under `perltie`, which makes a lot more sense. Moving `perldbmfilter` to its previous location is the only change I'd make here.

Thanks for your explanations.  I won't ask for any of the other things to be handled in this commit.  Okay by me to merge.

**Review by rwp0 — APPROVED** [review]

**Review by jkeenan — APPROVED** [review]

**Comment by jkeenan** [comment]
When I update blead after this merge and then try rebasing other branches on blead, I get a porting test failure.
```
orting/diag.t ............ ok                                          
===(   43850;2  1/?  51/?     0/1467  36/43  59/83 )====================
#   Failed test 'Is authors_file 'AUTHORS' up to date?'
#   at Porting/updateAUTHORS.pl line 127.
#   File 'AUTHORS' changes:
#     would add: mauke                          <lukasmai.403@gmail.com>
# 
# Files need updating! You probably just need to run
# 
#    Porting/updateAUTHORS.pl
# 
# and commit the results.
# Looks like you failed 1 test of 5.
porting/authors.t ......... Dubious, test returned 1 (wstat 256, 0x100) 
Failed 1/5 subtests 
```
Back in the day, I would have fixed this manually, noting that this p.r. has a different contributor name and email address from a previously known contributor.
```
mauke                          <lukasmai.403@gmail.com>

Lukas Mai                      <l.mai@web.de>
```
IIRC, it's always been our practice to have one contributor name able to be mapped to by multiple email addresses.  Can someone investigate?  Thanks.

**Comment by jkeenan** [comment]
> When I update blead after this merge and then try rebasing other branches on blead, I get a porting test failure.
> 

@mauke, I see that the name/email address from which you committed differs from the (older) name/email address in `AUTHORS`.
```
Author:     Lukas Mai <l.mai@web.de>
AuthorDate: Mon Sep 4 15:44:19 2023
Commit:     mauke <lukasmai.403@gmail.com>
CommitDate: Tue Sep 5 20:52:40 2023
```
@tonycoz, if you're around, please advise.  Thanks.

**Comment by mauke** [comment]
@jkeenan Weird. I didn't change anything on my end, but it's probably because of my shiny new commit bit.

I have a fix in 4d90825ea185bbada7c7525827d5830b44ce9206. I didn't realize it was already broken on blead. Should I just merge https://github.com/Perl/perl5/pull/21454 immediately?

**Comment by jkeenan** [comment]
On 9/5/23 17:40, mauke wrote:
> @jkeenan <https://github.com/jkeenan> Weird. I didn't change anything on 
> my end, but it's probably because of my shiny new commit bit.
> 
> I have a fix in 4d90825 
> <https://github.com/Perl/perl5/commit/4d90825ea185bbada7c7525827d5830b44ce9206>. I didn't realize it was already broken on blead. Should I just merge #21454 <https://github.com/Perl/perl5/pull/21454> immediately?
> 

Thanks for your rapid response.  Whoever bestowed that commit bit on you 
is probably more qualified to answer that question than I am! @tonycoz, 
if no one else.

jimk

**Pull request merged**
